### PR TITLE
Deploy Platform script. Minio Gateway Tweaks.

### PR DESCRIPTION
**Pull request by Misterturtle**
Updated the platform directory to properly deploy the platform with one command.

Edited Minio gateway to be more generic to work with google cloud in the future.

### CODE REVIEW

**Comment by JakeTheGingScheuer** [comment]
Just added in kdp and platform chart changes. Google credentials json file is now only created and volume mounted on deployment if minio gateway is gcs. Renamed some values for better clarity as well.

File: platform/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# To make changes to a dependent chart for development
+1. Make the changes you need on your local chart
+2. Navigate to the `docs/` directory and run `helm serve` and note the service address
+3. Open `requirements.yaml`
+4. Find the desired chart and change the value of `repository:` to `https://<SERVICE ADDRESS>`
+
+# Running in Azure Example
+`helm add repo https://smartcitiesdata.github.io/charts`
+```
+helm upgrade --install platform scdp/platform \
```
Comment (LtChae) [inline platform/README.md]: This would be more useful as a sample values file.

File: platform/templates/kafka/kafka.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      transaction.state.log.min.isr: 2
+    storage:
+      {{ if .Values.kafka.storage.enable }}
+      type: persistent-claim
+      size: .Values.kafka.storage.storageSize
+      deleteClaim: false
+      {{ else }}
+      type: ephemeral
+      {{ end }}
+    {{ if .Values.kafka.storage.enable }}
```
Comment (LtChae) [inline platform/templates/kafka/kafka.yaml]: Why does having storage enabled give kafka more cpu and ram? Is it needed to manage the volume?

File: platform/templates/secrets/ldap.yaml [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
   name: ldap
 type: Opaque
 data:
-  host: bGRhcA== #ldap
-  base_dn: ZGM9ZXhhbXBsZSxkYz1vcmc= #dc=example,dc=org
-  user: YWRtaW51c2VybmFtZT1hZG1pbg== #adminusername=admin
-  password: cGFzc3dvcmQ9YWRtaW4= #password=admin
-  environment_ou: dGVzdA== #test
-{{- end }}
+  host: bGRhcC1vcGVubGRhcA== # ldap-openldap
```
Comment (LtChae) [inline platform/templates/secrets/ldap.yaml]: We should consider making this section more configurable and secure. Probably out of scope here.

File: platform/requirements.yaml [hunk 4]
```diff
@@ -5,41 +5,41 @@ dependencies:
     condition: openldap.enabled
   - name: andi
     version: 1.1.3
-    repository: https://smartcitiesdata.github.io/charts
+    repository: https://raw.githubusercontent.com/PillarTechnology/charts-1/master/docs
```
Comment (ManApart) [inline platform/requirements.yaml]: This doesn't seem like the right url for this.

File: reaper/values.yaml [hunk 5]
```diff
@@ -1,9 +1,11 @@
 outputTopicPrefix: raw
 dlqTopic: streaming-dead-letters
 replicaCount: 1
+metadata:
+  labels:
 image:
-  repository: smartcitiesdata/reaper
-  tag: development
+  repository: rconaway2/reaper
```
Comment (ManApart) [inline reaper/values.yaml]: Same here, looks like they're pointing to a fork?

File: reaper/values.yaml [hunk 6]
```diff
@@ -1,9 +1,11 @@
...
+  tag: 86.0.0
```
Comment (ManApart) [inline reaper/values.yaml]: This should still be dev.

**Comment by LtChae** [comment]
Update the description on this to omit the redis work, as this is now merged into the main platform.

File: platform/requirements.yaml [hunk 7]
```diff
@@ -1,45 +1,45 @@
 dependencies:
+  - name: openldap
+    version: 1.2.2
+    repository: https://kubernetes-charts.storage.googleapis.com/
+    condition: openldap.enabled
   - name: andi
-    version: 1.1.2
-    repository: https://smartcitiesdata.github.io/charts
+    version: 1.1.3
+    repository: https://raw.githubusercontent.com/PillarTechnology/charts-1/master/docs
```
Comment (jeffgrunewald) [inline platform/requirements.yaml]:
```suggestion
    repository: https://smartcitiesdata.github.io/charts
```

File: platform/requirements.yaml [hunk 8]
```diff
@@ -1,45 +1,45 @@
...
   - name: andi
-    version: 1.1.2
+    version: 1.1.3
     repository: https://smartcitiesdata.github.io/charts
     condition: andi.enabled
   - name: discovery-api
-    version: 0.4.2
-    repository: https://smartcitiesdata.github.io/charts
+    version: 0.4.4
+    repository: https://raw.githubusercontent.com/PillarTechnology/charts-1/master/docs
```
Comment (jeffgrunewald) [inline platform/requirements.yaml]:
```suggestion
    repository: https://smartcitiesdata.github.io/charts
```

File: platform/requirements.yaml [hunk 9]
```diff
@@ -1,45 +1,45 @@
...
     condition: andi.enabled
   - name: discovery-api
-    version: 0.4.2
+    version: 0.4.4
     repository: https://smartcitiesdata.github.io/charts
     condition: discovery-api.enabled
   - name: discovery-streams
     version: 0.1.4
-    repository: https://smartcitiesdata.github.io/charts
+    repository: https://raw.githubusercontent.com/PillarTechnology/charts-1/master/docs
```
Comment (jeffgrunewald) [inline platform/requirements.yaml]:
```suggestion
    repository: https://smartcitiesdata.github.io/charts
```

File: reaper/values.yaml (hunk 5 above)
Comment (jeffgrunewald) [inline reaper/values.yaml]:
```suggestion
  repository: smartcitiesdata/reaper
```

File: reaper/values.yaml (hunk 6 above)
Comment (jeffgrunewald) [inline reaper/values.yaml]:
```suggestion
  tag: development
```

File: platform/requirements.yaml [hunk 10]
```diff
@@ -1,45 +1,45 @@
...
   - name: flair
     version: 2.1.0
-    repository: https://smartcitiesdata.github.io/charts
+    repository:  https://smartcitiesdata.github.io/charts
     condition: flair.enabled
   - name: forklift
-    version: 2.4.0
-    repository: https://smartcitiesdata.github.io/charts
+    version: 2.4.1
+    repository: https://raw.githubusercontent.com/PillarTechnology/charts-1/master/docs
```
Comment (jeffgrunewald) [inline platform/requirements.yaml]:
```suggestion
    repository: https://smartcitiesdata.github.io/charts
```

File: platform/requirements.yaml [hunk 11]
```diff
@@ -1,45 +1,45 @@
...
     condition: forklift.enabled
   - name: kubernetes-data-platform
-    version: 1.0.0
-    repository: https://smartcitiesdata.github.io/charts
-    condition: kdp.enabled
+    version: 1.1.2
+    repository: https://raw.githubusercontent.com/PillarTechnology/charts-1/master/docs
```
Comment (jeffgrunewald) [inline platform/requirements.yaml]:
```suggestion
    repository: https://smartcitiesdata.github.io/charts
```

File: platform/requirements.yaml [hunk 12]
```diff
@@ -1,45 +1,45 @@
...
+    condition: kubernetes-data-platform.enabled
   - name: odo
     version: 0.2.2
     repository: https://smartcitiesdata.github.io/charts
     condition: odo.enabled
   - name: reaper
-    version: 0.1.1
-    repository: https://smartcitiesdata.github.io/charts
+    version: 0.1.3
+    repository: https://raw.githubusercontent.com/PillarTechnology/charts-1/master/docs
```
Comment (jeffgrunewald) [inline platform/requirements.yaml]:
```suggestion
    repository: https://smartcitiesdata.github.io/charts
```

File: platform/requirements.yaml [hunk 13]
```diff
@@ -1,45 +1,45 @@
...
   - name: redis
-    version: 8.0.1
+    version: 9.5.0
     repository: https://kubernetes-charts.storage.googleapis.com/
     condition: redis.enabled
   - name: valkyrie
-    version: 2.4.0
-    repository: https://smartcitiesdata.github.io/charts
+    version: 2.4.1
+    repository: https://raw.githubusercontent.com/PillarTechnology/charts-1/master/docs
```
Comment (jeffgrunewald) [inline platform/requirements.yaml]:
```suggestion
    repository: https://smartcitiesdata.github.io/charts
```

File: platform/requirements.yaml [hunk 14]
```diff
@@ -1,45 +1,45 @@
...
     repository: https://smartcitiesdata.github.io/charts
     condition: discovery-streams.enabled
-  - name: discovery-ui
-    version: 0.2.3
-    repository: https://smartcitiesdata.github.io/charts
-    condition: discovery-ui.enabled
   - name: flair
     version: 2.1.0
-    repository: https://smartcitiesdata.github.io/charts
+    repository:  https://smartcitiesdata.github.io/charts
```
Comment (jeffgrunewald) [inline platform/requirements.yaml]:
```suggestion
    repository: https://smartcitiesdata.github.io/charts
```

File: platform/templates/kafka/topics.yaml [hunk 15]
```diff
@@ -1,19 +1,22 @@
...
 ---
 apiVersion: kafka.strimzi.io/v1alpha1
 kind: KafkaTopic
 metadata:
-  name: {{ $topic.name }}
-  namespace: {{ $topic.namespace | default $.Release.Namespace }}
+  name: streaming-dead-letters
   labels:
-    strimzi.io/cluster: {{ $topic.strimziCluster | default $.Values.kafka.strimziCluster }}
+    strimzi.io/cluster: streaming-service
```
Comment (jeffgrunewald) [inline platform/templates/kafka/topics.yaml]:
```suggestion
    strimzi.io/cluster: {{ .Values.kafka.strimziCluster }}
```

File: platform/templates/kafka/topics.yaml [hunk 16]
```diff
@@ -1,19 +1,22 @@
...
   config: {}
-{{- end }}
-{{- end }}
+---
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: KafkaTopic
+metadata:
+  name: event-stream
+  labels:
+    strimzi.io/cluster: streaming-service
```
Comment (jeffgrunewald) [inline platform/templates/kafka/topics.yaml]:
```suggestion
    strimzi.io/cluster: {{ .Values.kafka.strimziCluster }}
```

**Comment by jeffgrunewald** [comment]
Merged in under a different PR

**Pull request closed without merging**